src/recorder.js

```diff
@@ -0,0 +1,53 @@
+var remix = require('ethereum-remix')
+var EventManager = remix.lib.EventManager
+
+class Recorder {
+  constructor (opts = {}) {
+    var self = this
+    self._api = opts.api
+    self.event = new EventManager()
+    self.data = { journal: [], _pending: {} }
+    opts.events.executioncontext.register('contextChanged', function () {
+      self.clearAll()
+    })
+    var counter = 1
+    self._addressCache = {}
+    opts.events.udapp.register('initiatingTransaction', (stamp, tx) => {
+      var { from, to, value, gas, data } = tx
+      var record = { value, gas, data }
+      record.from = self._addressCache[from] || (self._addressCache[from] = `<account -${(++counter)}>`)
+      if (to === null) self.data._pending[stamp] = record
+      else record.to = self._addressCache[to] || (self._addressCache[to] = `<account -${(++counter)}>`)
+      self.append(stamp, record)
+    })
+    opts.events.udapp.register('transactionExecuted', args => {
```

Comment:
this event has more than one arg:
`transactionExecuted', [error, args.from, args.to, args.data, false, result, stamp]`